packages/clerk-js/src/v4/primitives/Input.tsx

```diff
@@ -0,0 +1,44 @@
+import { createVariants, PrimitiveProps, StyleVariants } from '../styledSystem';
+import { useInput } from './hooks/useInput';
+
+// TODO: Connect with CSS variables
+
+const { applyVariants, filterProps } = createVariants((theme, props) => ({
+  base: {
+    margin: 0,
+    outline: 0,
+    padding: `${theme.sizes.$2x5} ${theme.sizes.$4}`,
+    fontStyle: 'normal',
+    fontWeight: theme.fontWeights.$normal,
+    border: '1px solid',
+    borderColor: props.hasError ? theme.colors.$danger500 : theme.colors.$blackAlpha300,
+    borderRadius: theme.radii.$md,
+    '&:focus-visible': {
+      borderColor: props.hasError ? theme.colors.$danger500 : theme.colors.$primary50,
+      boxShadow: 'none',
+    },
+  },
+  variants: {},
+}));
+
+type InputProps = PrimitiveProps<'input'> &
+  StyleVariants<typeof applyVariants> & {
+    isDisabled?: boolean;
+    hasError?: boolean;
+  };
+
+const Input = (props: InputProps): JSX.Element => {
```

Comment:
❓ This might need to use a `React.forwardRef` like
```
const Button = React.forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => {
```
but we can always add it later

Reply:
Added